### ISSUE TITLE
Require a community when uploading

### ISSUE THREAD

**Issue by Samk13** [inveniosoftware/product-rdm#183]
## Is your feature request related to a problem?
Organizations may require at least one community to be associated with a record. Currently, there is no configuration option to enforce this constraint.

## Describe the solution you'd like
- Add a new configuration option that prevents users from deleting all communities in a record, except for the last one. By default, this feature should be off and can be activated in an instance by adding the following configuration:
```python
RDM_RECORD_ALWAYS_IN_COMMUNITY = True
```
- When this configuration is set to True, users should still be able to remove communities, but not the last one, ensuring that there is always at least one community associated with the record.
- Assert that users are unable to directly publish a record without going through a community submit process. In this scenario, the community manager would be responsible for publishing the record. This can be achieved by disabling the "publish" button based on the configuration value and ensuring that users can only submit the record for review to a community manager.
This issue:
closes https://github.com/inveniosoftware/invenio-rdm-records/pull/1719
closes inveniosoftware/invenio-app-rdm#2642
resolves inveniosoftware/invenio-app-rdm#2201
resolves inveniosoftware/react-invenio-forms#202

## Describe alternatives you've considered
An alternative solution could involve implementing a more granular permission system that allows administrators to define and enforce specific rules for community association. However, this would require a more complex implementation and might introduce unnecessary complexity for most use cases.

## Additional context
This feature request is related to the issue of disabling the "publish" button based on the value of a config variable and raising validation errors if the user hasn't selected a community when publishing. Implementing this configuration option will provide organizations with more flexibility in managing their records and community associations, as well as ensuring that the community managers have control over the publishing process.

This feature is related to CERNDocumentServer/cds-rdm#55 
We can bring up the topic of introducing custom roles in place of granting full admin access for community creation. As community managers, we're interested in limiting their permissions to manage their respective communities only, rather than having complete control over the system. We discuss this further and explore potential solutions in separate issues when this one is merged.

**Comment by tzubaki** [comment]
For JRC, this feature is needed for the first production release, as we should not allow people to publish out of communities. The solution proposed by Nicola, when the button Publish is still enable, but a meaningfull error message will prevent from publishing is ok. Could this still be included in V12?

**Comment by ntarocco** [comment]
> For JRC, this feature is needed for the first production release, as we should not allow people to publish out of communities. The solution proposed by Nicola, when the button Publish is still enable, but a meaningfull error message will prevent from publishing is ok. Could this still be included in V12?

If not included in v12, it will be part of a minor release, e.g. v12.1

**Comment by Samk13** [comment]
@ntarocco Great progress so far, but should we reopen this issue because the `community-creator` role has not been enforced. 
While the [merged PR](https://github.com/inveniosoftware/invenio-rdm-records/pull/1813) implements the requirement for records to be associated with a community before publishing, it does not address the need for a distinct role to control who can create communities see example [here](https://github.com/Samk13/rdm-restricted-permissions/blob/master/site/kth_rdm_v12/permissions.py#L23-L31). As of now, any user can still create a community and submit records, which does not meet the original request.

**Comment by ntarocco** [comment]
> @ntarocco Great progress so far, but should we reopen this issue because the `community-creator` role has not been enforced. While the [merged PR](https://github.com/inveniosoftware/invenio-rdm-records/pull/1813) implements the requirement for records to be associated with a community before publishing, it does not address the need for a distinct role to control who can create communities see example [here](https://github.com/Samk13/rdm-restricted-permissions/blob/master/site/kth_rdm_v12/permissions.py#L23-L31). As of now, any user can still create a community and submit records, which does not meet the original request.

This can be achieved simply overriding the service permissions, see an example [here](https://github.com/CERNDocumentServer/cds-rdm/blob/master/site/cds_rdm/permissions.py#L24).

**Comment by Samk13** [comment]
@ntarocco Thank you for your response and for pointing me to the example of overriding service permissions. 
I understand that we can restrict who can create communities by customizing permissions at the instance level. However, I believe we should reconsider and address this issue within the core codebase for a few reasons:

User Interface Consistency: Even if we restrict permissions, users still have access to the community settings page where they can attempt to change options like flipping a community from closed to open. While the permissions will prevent the action see [here](https://github.com/inveniosoftware/invenio-communities/pull/1231), having the option visible can lead to confusion and a less streamlined user experience. Removing or conditionally displaying these options based on permissions would enhance usability.

Future Maintainability: By integrating conditional code directly into the communities module, we ensure that future changes will naturally adapt to this mode. It's easy for developers to overlook instance-specific permission overrides during updates, which could lead to conflicts or security issues down the line. Embedding this logic in the core codebase makes it more robust and less error-prone.

Alignment with Original Requirements: The original request emphasized the need for a distinct role controlling who can create communities, separate from the admin role. While instance-level permission overrides provide a workaround, they don't fully meet the requirement of having this role managed within the system's core functionality. Implementing this directly addresses the initial use case more effectively.

I understand that this might require additional work now, but it could prevent multiple conflicts and issues in the future. Given the active development of the communities module with subcommunities etc..., incorporating this case would ensure that the system remains adaptable and secure as it evolves.

I'm more than willing to assist with the necessary changes to implement this functionality. Let's work together to enhance the platform for all users. Please let me know your thoughts on this approach.

**Comment by ntarocco** [comment]
@Samk13 there is one point that is not very clear to me:

> Even if we restrict permissions, users still have access to the community settings page where they can attempt to change options like flipping a community from closed to open.

What is the exact use case? If I can create a community, and I therefore owner, I should be able to adjust its visibility too.

I checked the [PR](https://github.com/inveniosoftware/invenio-communities/pull/1231), it feels like that we are mixing the feature to submit a record without a review with the community required flag.

**Comment by Samk13** [comment]
@ntarocco Isn't the whole point of requiring "community on publish" to enforce a review process for each record?

It looks like your GPT summary didn’t accurately capture my Copilot AI comment ;)  so to clarify, I’m not talking about visibility settings here,  I’m specifically suggesting that we set the "[review policy](https://github.com/inveniosoftware/invenio-communities/blob/8c0e30f161f3ba22f8febe6db692cb546d4ea7c3/invenio_communities/views/communities.py#L87-L110)" to "closed," keeping the focus on enforcing a review process.

As it seems like you see the "community creator" role is currently treated as an admin, but this shouldn’t be the case. they should still follow the organization’s publish under review rule.

I also invite others here to share their thoughts on this, as this feature has been requested by many other organizations not only us.

**Comment by ntarocco** [comment]
> @ntarocco Isn't the whole point of requiring "community on publish" to enforce a review process for each record?

I don't think that we should mix and merge these 2. I think that it is a realistic use case to require a community, so you are sure that community curators can curate all records in your repository. And, at the same time, allow community owners to decide if each submission requires a review or not.
The auto-inclusion is a community setting, a community policy. The community required feature is a global system setting.

If your instance requires these 2 together, then we can discuss what is the best way to implement this in your instance.

We should keep InvenioRDM simple, understandable. This also means keeping under control the amount of possible available configurations.

I re-opened the discussion and it would be indeed nice to have other users expressing their opinions on this topic ☺️

**Comment by mugraph** [comment]
This may be slightly OT but I felt I may add this observation here while the issue is re-opened.

Trying out the current state of development on this feature, I came to wonder if the `RDM_RECORD_ALWAYS_IN_COMMUNITY = True`  config should not also disable (or hide) the "Publish without community" button that is shown for declined drafts, as this may result in confusion on the user side with no clear way to recover, besides the "Change community" option.

**Comment by ntarocco** [comment]
> This may be slightly OT but I felt I may add this observation here while the issue is re-opened.
> 
> Trying out the current state of development on this feature, I came to wonder if the `RDM_RECORD_ALWAYS_IN_COMMUNITY = True` config should not also disable (or hide) the "Publish without community" button that is shown for declined drafts, as this may result in confusion on the user side with no clear way to recover, besides the "Change community" option.

@mugraph the answer to this question is [up here](https://github.com/inveniosoftware/product-rdm/issues/183#issuecomment-2025330231) in this conversation (already many comments :)): the short answer is that, at the moment, there is no easy way to implement this in InvenioRDM.
